### ISSUE TITLE
Stable Diffusion XL Pipeline generator argument is not deterministic

### ISSUE THREAD

**Issue by CoffeeVampir3** [huggingface/diffusers#4483]
### Describe the bug

Passing a torch seeded generator
```
generator = torch.Generator("cuda")
generator.manual_seed(1337)
```

Does not seem to matter when passed to the SDXL pipe, the generator seems to be completely random. I did some digging but could not find the root cause of the issue.

### Reproduction

Exact reproduction:
```
import torch
from diffusers import StableDiffusionXLPipeline
from PIL import Image
StableDiffusionXLPipeline.from_pretrained("stabilityai/stable-diffusion-xl-base-1.0", torch_dtype=torch.bfloat16, use_safetensors=True, variant="fp16", add_watermarker=False)
generator = torch.Generator("cuda")
generator.manual_seed(1337)
images = pipe(prompt = ["Cute Puppy"] * 4, generator=generator).images
```

to display in jupyter-notebook:
```
from PIL import Image
import matplotlib.pyplot as plt
from IPython.display import display
fig, axs = plt.subplots(nrows=2, ncols=2, figsize=(10,10))
for img, ax in zip(images, axs.flatten()):
    ax.imshow(img)
    ax.axis('off')
plt.show()
```

Run the pipe twice, the expected result would be to get the exact same images twice, but instead every run generates different unique images.

### System Info

- `diffusers` version: 0.20.0.dev0
- Platform: Linux-6.4.8-arch1-1-x86_64-with-glibc2.38
- Python version: 3.10.12
- PyTorch version (GPU?): 2.1.0.dev20230718 (True)
- Huggingface_hub version: 0.15.1
- Transformers version: 4.32.0.dev0
- Accelerate version: 0.22.0.dev0
- xFormers version: 0.0.21+2d3a221.d20230804
- Using GPU in script?: cuda
- Using distributed or parallel set-up in script?: no

### Who can help?

@patrickvonplaten @sayakpaul @williamberman

**Comment by xiankgx** [comment]
I don't think this is a bug. You should just reset the manual seed in the generator prior to each generation runs.

**Comment by sayakpaul** [comment]
Yes, you should just reinitialize the generator and then see what happens. Closing it hence, feel free to re-open in case the suggestions don't solve the problem.
